build_tools/setup_helpers/extension.py

```diff
@@ -1,189 +1,120 @@
+# _LIBTORCHTEXT_NAME = "torchtext.lib.libtorchtext"
+_EXT_NAME = "torchtext._torchtext"
+_THIS_DIR = Path(__file__).parent.resolve()
+_ROOT_DIR = _THIS_DIR.parent.parent.resolve()
 
-def _configure_third_party(debug):
-    _build_third_party(debug)
-    _build_sentence_piece(debug)
 
+def get_ext_modules():
+    modules = [
+        # TODO: Following line will be uncommented when adding splitting up the cpp libraries to `libtorchtext` and `_torchtext`
+        # Extension(name=_LIBTORCHTEXT_NAME, sources=[]),
+        Extension(name=_EXT_NAME, sources=[]),
+    ]
+    return modules
 
-_EXT_NAME = "torchtext._torchtext"
 
+# Based off of
+# https://github.com/pybind/cmake_example/blob/580c5fd29d4651db99d8874714b07c0c49a53f8a/setup.py
 
-def get_ext_modules(debug=False):
-    return [
-        CppExtension(
-            _EXT_NAME,
-            _get_srcs(),
-            libraries=_get_libraries(),
-            include_dirs=_get_include_dirs(),
-            library_dirs=_get_library_dirs(),
-            extra_compile_args=_get_eca(debug),
-            extra_link_args=_get_ela(debug),
-        ),
-    ]
 
+class CMakeBuild(build_ext):
+    def run(self):
+        try:
+            subprocess.check_output(["cmake", "--version"])
+        except OSError:
+            raise RuntimeError("CMake is not available.") from None
+        super().run()
 
-class BuildExtension(TorchBuildExtension):
     def build_extension(self, ext):
-        if ext.name == _EXT_NAME:
-            _configure_third_party(self.debug)
-        super().build_extension(ext)
+        # Since two library files (libtorchaudio and _torchaudio) need to be
+        # recognized by setuptools, we instantiate `Extension` twice. (see `get_ext_modules`)
+        # This leads to the situation where this `build_extension` method is called twice.
+        # However, the following `cmake` command will build all of them at the same time,
+        # so, we do not need to perform `cmake` twice.
+        # Therefore we call `cmake` only for `torchaudio._torchaudio`.
+        if ext.name != "torchtext._torchtext":
+            return
+
+        extdir = os.path.abspath(os.path.dirname(self.get_ext_fullpath(ext.name)))
+
+        # required for auto-detection of auxiliary "native" libs
+        if not extdir.endswith(os.path.sep):
+            extdir += os.path.sep
+
+        cfg = "Debug" if self.debug else "Release"
+
+        cmake_args = [
+            f"-DCMAKE_BUILD_TYPE={cfg}",
+            f"-DCMAKE_PREFIX_PATH={torch.utils.cmake_prefix_path}",
+            f"-DCMAKE_INSTALL_PREFIX={extdir}",
+            "-DCMAKE_VERBOSE_MAKEFILE=ON",
+            f"-DPython_INCLUDE_DIR={distutils.sysconfig.get_python_inc()}",
+            "-DBUILD_TORCHTEXT_PYTHON_EXTENSION:BOOL=ON",
+            "-DRE2_BUILD_TESTING:BOOL=OFF",
+            "-DBUILD_TESTING:BOOL=OFF" "-DBUILD_SHARED_LIBS=OFF",
+            "-DCMAKE_POLICY_DEFAULT_CMP0063=NEW",
+            "-DCMAKE_CXX_FLAGS=" + _get_cxx11_abi(),
```

Comment:
This is supposed to be handled by TORCH_CXX_FLAGS.

However, https://github.com/pytorch/audio/issues/2287 suggests this does not work with manually built PyTorch, so I will follow up with this one.

Reply:
Gotcha, I will keep this in for now until the issue is resolved then.